application/apps/rustcore/rs-bindings/src/js/jobs/converting.rs

```diff
@@ -0,0 +1,179 @@
+use command_outcome::Cancelled;
+use node_bindgen::{
+    core::{val::JsEnv, NjError, TryIntoJs},
+    sys::napi_value,
+};
+use output::{Empty, StringVec};
+use proto::*;
+use session::unbound::commands::CommandOutcome;
+
+pub(crate) struct CommandOutcomeWrapper<T>(pub Option<CommandOutcome<T>>);
+
+impl<T> CommandOutcomeWrapper<T> {
+    pub fn new(val: CommandOutcome<T>) -> Self {
+        CommandOutcomeWrapper(Some(val))
+    }
+}
+
+fn get<T>(mut output: Option<CommandOutcome<T>>) -> Option<T> {
+    match output.take().expect("Command output has to be provided") {
+        CommandOutcome::Finished(output) => Some(output),
+        CommandOutcome::Cancelled => None,
+    }
+}
+
+impl From<CommandOutcomeWrapper<String>> for Vec<u8> {
+    fn from(output: CommandOutcomeWrapper<String>) -> Self {
+        let msg = get(output.0)
+            .map(|v| proto::CommandOutcome {
+                outcome: Some(command_outcome::Outcome::Finished(
+                    command_outcome::Finished {
+                        result: Some(Output {
+                            output: Some(output::Output::StringValue(v)),
+                        }),
+                    },
+                )),
+            })
+            .unwrap_or(proto::CommandOutcome {
+                outcome: Some(command_outcome::Outcome::Cancelled(Cancelled {})),
+            });
```

Comment:
If these parts are performance-critical then, I would try if using `unwrap_or_else()` here would bring any improvements because of its lazy loading. Although I don't think it would matter that much since the structs inside the unwrap function aren't that heavy

Reply:
nice work Dmitry!
until we have figured out the performance part, please mark it as "Draft"